src/main/java/ca/fxco/pistonlib/base/ModItems.java

```diff
@@ -0,0 +1,84 @@
+    public static final BlockItem BASIC_PISTON = registerBlock(ModBlocks.BASIC_PISTON);
+    public static final BlockItem BASIC_STICKY_PISTON = registerBlock(ModBlocks.BASIC_STICKY_PISTON);
+    public static final BlockItem LONG_PISTON = registerBlock(ModBlocks.LONG_PISTON);
+    public static final BlockItem LONG_STICKY_PISTON = registerBlock(ModBlocks.LONG_STICKY_PISTON);
+    public static final BlockItem STALE_PISTON = registerBlock(ModBlocks.STALE_PISTON);
+    public static final BlockItem STALE_STICKY_PISTON = registerBlock(ModBlocks.STALE_STICKY_PISTON);
+    public static final BlockItem VERY_QUASI_PISTON = registerBlock(ModBlocks.VERY_QUASI_PISTON);
+    public static final BlockItem VERY_QUASI_STICKY_PISTON = registerBlock(ModBlocks.VERY_QUASI_STICKY_PISTON);
+    public static final BlockItem STRONG_PISTON = registerBlock(ModBlocks.STRONG_PISTON);
+    public static final BlockItem STRONG_STICKY_PISTON = registerBlock(ModBlocks.STRONG_STICKY_PISTON);
+    public static final BlockItem FAST_PISTON = registerBlock(ModBlocks.FAST_PISTON);
+    public static final BlockItem FAST_STICKY_PISTON = registerBlock(ModBlocks.FAST_STICKY_PISTON);
+    public static final BlockItem VERY_STICKY_PISTON = registerBlock(ModBlocks.VERY_STICKY_PISTON);
+    public static final BlockItem FRONT_POWERED_PISTON = registerBlock(ModBlocks.FRONT_POWERED_PISTON);
+    public static final BlockItem FRONT_POWERED_STICKY_PISTON = registerBlock(ModBlocks.FRONT_POWERED_STICKY_PISTON);
+    public static final BlockItem TRANSLOCATION_PISTON = registerBlock(ModBlocks.TRANSLOCATION_PISTON);
+    public static final BlockItem TRANSLOCATION_STICKY_PISTON = registerBlock(ModBlocks.TRANSLOCATION_STICKY_PISTON);
+    public static final BlockItem SLIPPERY_PISTON = registerBlock(ModBlocks.SLIPPERY_PISTON);
+    public static final BlockItem SLIPPERY_STICKY_PISTON = registerBlock(ModBlocks.SLIPPERY_STICKY_PISTON);
+    public static final BlockItem SUPER_PISTON = registerBlock(ModBlocks.SUPER_PISTON);
+    public static final BlockItem SUPER_STICKY_PISTON = registerBlock(ModBlocks.SUPER_STICKY_PISTON);
+    public static final BlockItem MBE_PISTON = registerBlock(ModBlocks.MBE_PISTON);
+    public static final BlockItem MBE_STICKY_PISTON = registerBlock(ModBlocks.MBE_STICKY_PISTON);
+    public static final BlockItem MERGE_PISTON = registerBlock(ModBlocks.MERGE_PISTON);
+    public static final BlockItem MERGE_STICKY_PISTON = registerBlock(ModBlocks.MERGE_STICKY_PISTON);
+
+    private static BlockItem registerBlock(Block block) {
+        return register(BuiltInRegistries.BLOCK.getKey(block), new BlockItem(block, new Item.Properties()));
+    }
+
+    private static <T extends Item> T register(String name, T item) {
+        return register(id(name), item);
+    }
+
+    private static <T extends Item> T register(ResourceLocation id, T item) {
+        ItemGroupEvents.modifyEntriesEvent(ModCreativeModeTabs.GENERAL).register(content -> content.accept(item));
```

Comment:
This should be removed and done in the `ModCreativeModeTabs` class

Reply:
As fun as that sounds, I don't think we should be manually adding all the items. 
The more tasks I have to do when I want to add a single block, the less im going to want to do it.